src/Components/FormCreation/FormCard.js

```diff
@@ -127,142 +172,296 @@ function FormCard({
   handleDuplicate,
   handleDelete,
   sectionKey,
-  questionKey
+  questionKey,
+  handleQuestionDescriptionUpdate,
+  handleQuestionTitleUpdate,
+  handleQuestionTypeUpdate,
+  handleQuestionContentUpdate,
+  handleRequiredToggle
 }) {
   const classes = useStyles();
-  const [required, setRequired] = useState(card.required);
   const [title, setTitle] = useState(card.name);
   const [description, setDescription] = useState(card.description);
-  const [questionMenuAnchor, setQuestionMenuAnchor] = useState(false);
+  const [questionMenuAnchor, setQuestionMenuAnchor] = useState(null);
 
-  const handleSwitch = () => {
-    setRequired((prev) => !prev);
+  const onQuestionUpdate = (options) => {
+    if (card.type === "CHECKBOXES" || card.type === "MULTIPLE_CHOICE") {
+      const opt = {
+        xoptions: options.map((option) => option[0]),
+        yoptions: null
+      };
+      handleQuestionContentUpdate(opt);
+    }
+    if (card.type === "CHECKBOX_GRID") {
+      //todo
+    }
+    if (card.type === "FILE_UPLOAD") {
+      //todo
+    }
   };
 
+  const questionTypes = [
+    {
```

Comment:
This const is iterated over 3 times during render. Once for menu button title, loading menu items and rendering the correct card contents

Saved a lot of code (change not reflected in submission version)